pkg/networkservice/common/switchcase/client.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package switchcase
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+)
+
+// ClientCase is a case type for the switch-case client chain element
+type ClientCase struct {
+	Condition Condition
```

Comment:
Can we remove `Default` condition and  look at nil condition as a Default?

For example,
```go
type ClientCase struct{
 Client: ... // Note: Condition is nil
}
```

This means call `Client` with default condition.

Thoughts?

Reply:
Yes, looks great.

Reply:
One question... is 'nil' clearer than Default ?

It might make more sense semantically to keep the 'Default' condition as something we can list...